// @fluentui/react-portal/lib-commonjs/components/Portal/usePortal.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    toMountNodeProps: function() {
        return toMountNodeProps;
    },
    usePortal_unstable: function() {
        return usePortal_unstable;
    }
});
const _interop_require_wildcard = require("@swc/helpers/_/_interop_require_wildcard");
const _reactutilities = require("@fluentui/react-utilities");
const _react = /*#__PURE__*/ _interop_require_wildcard._(require("react"));
const _usePortalMountNode = require("./usePortalMountNode");
function toMountNodeProps(mountNode) {
    if ((0, _reactutilities.isHTMLElement)(mountNode)) {
        return {
            element: mountNode
        };
    }
    if (typeof mountNode === 'object') {
        if (mountNode === null) {
            return {
                element: null
            };
        }
        return mountNode;
    }
    return {};
}
const usePortal_unstable = (props)=>{
    const { element, className } = toMountNodeProps(props.mountNode);
    const virtualParentRootRef = _react.useRef(null);
    const fallbackElement = (0, _usePortalMountNode.usePortalMountNode)({
        disabled: !!element,
        className
    });
    const state = {
        children: props.children,
        mountNode: element !== null && element !== void 0 ? element : fallbackElement,
        virtualParentRootRef
    };
    _react.useEffect(()=>{
        if (state.virtualParentRootRef.current && state.mountNode) {
            (0, _reactutilities.setVirtualParent)(state.mountNode, state.virtualParentRootRef.current);
        }
        return ()=>{
            if (state.mountNode) {
                (0, _reactutilities.setVirtualParent)(state.mountNode, undefined);
            }
        };
    }, [
        state.virtualParentRootRef,
        state.mountNode
    ]);
    return state;
};
